src/main/java/org/skriptlang/skript/lang/experiment/ExperimentRegistry.java

```diff
@@ -0,0 +1,106 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.lang.experiment;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptAddon;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+/**
+ * A manager for registering (and identifying) experimental feature flags.
+ */
+/*
+* TODO
+* 	This is designed to be (replaced by|refactored into) a proper registry when the registries rework PR
+* 	is completed. The overall skeleton is designed to remain, so that there should be no breaking changes
+* 	for anything using it. I.e. you will still be able to use Skript#experiments() and obtain 'this' class
+* 	although these will just become helper methods for the proper registry behaviour.
+* */
+public class ExperimentRegistry implements Experimented {
+
+	private final Skript skript;
+	private final Set<Experiment> experiments;
+
+	public ExperimentRegistry(Skript skript) {
+		this.skript = skript;
+		this.experiments = new LinkedHashSet<>();
+	}
+
+	/**
+	 * Finds an experiment matching this name. If none exist, an 'unknown' one will be created.
+	 * @param text The text provided by the user.
+	 * @return An experiment.
+	 */
+	public @NotNull Experiment find(String text) {
+		if (experiments.isEmpty())
+			return Experiment.unknown(text);
+		for (Experiment experiment : experiments) {
+			if (experiment.matches(text))
+				return experiment;
+		}
+		return Experiment.unknown(text);
+	}
+
+	/**
+	 * @return All currently-registered experiments.
+	 */
+	public Experiment[] registered() {
+		return experiments.toArray(new Experiment[0]);
+	}
+
+	/**
+	 * Registers a new experimental feature flag, which will be available to scripts
+	 * with the {@code using %name%} structure.
+	 * @param addon The source of this feature.
+	 * @param experiment The experimental feature flag.
+	 */
+	public void register(SkriptAddon addon, Experiment experiment) {
```

Comment:
I'd like to see a way to unregister experiments.

Reply:
I thought your registries PR was going to handle this?

Reply:
We could implement this later (when registries are merged), but I do not see any reason not to do it now.